### PR TITLE
Display pages as a nested list

### PR DESCRIPTION
**Pull request by Dignifiedquire**
Basics are working, mostly bug fixing needed to do

screen shot 2015-09-17 at 19 37 27

Todo:

- [x] Handle reload selection
- [x] Display headings in the sidebar

### CODE REVIEW

File: lib/ex_doc/formatter/html/templates/sidebar_template.eex [hunk 1]
```diff
@@ -27,15 +27,11 @@
     <input type="text" class="sidebar-searchInput" placeholder="search" autocomplete="off" results="0" />
   </div>
 
-  <ul class="sidebar-mainNav">
```
Comment (josevalim) [inline lib/ex_doc/formatter/html/templates/sidebar_template.eex]: Don't forget to remove the styles for this.

**Comment by Dignifiedquire** [comment]
@josevalim please review and merge if you are happy

**Comment by eksperimental** [comment]
please hold it for a bit. do not merge, until i the PR with renaming the elements is merged. Since i had to rebase and there is already an extra commit,. I think to avoid polluting the commit history, it will be best to wait for that

**Comment by josevalim** [comment]
I am merging this locally and I will push.

**Comment by josevalim** [comment]
@Dignifiedquire can we make the h2 under the page name have the `|` and `>` as we do for functions, types and macros under modules or it would be too complicated here?

**Comment by josevalim** [comment]
Other than that, this looks great! :D

**Comment by eksperimental** [comment]
sorry but what is that capital letter underscore thing? like E_views ?

**Comment by josevalim** [comment]
@eksperimental it is how the Phoenix guides are organized. Nothing we can do here. :)

**Comment by eksperimental** [comment]
@josevalim i was puzzled, i thought that was part of what this new feature was introducing.

**Comment by josevalim** [comment]
fortunately it isn't :D

**Pull request merged**